### ISSUE TITLE
[BUG] spooled / recipe detected tile overviews queuing shifted ROIs

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#648]
**Describe the bug**
Running a tile using the tiler panel, followed by a detection recipe in the tile viewer and `Queue from tile viewer` works, however running a tile protocol spooled with a chained detection recipe (same recipe with the addition of supertile module) and `recipes.acquisition.QueueAcquisitions`) gets bogus ROIs which are presumably shifted. Note that said recipe also saves the ROIs to disk, and `Queue ROIs from file` also does not pull legitimate ROIs (which is interesting given the difference between centre_roi_on vs. positioning.update between QueueAcquisitions and the actionUI GUI methods.

**Comment by barentine** [comment]
Current status is unresolved:
two scans, circular, each 200 um radius (setting, anyway)
Both were started at (x, y) = 3500, 43200 um, with the multiview ROI already as it was for the scans.

**metadata from tiler ImagePyramid**
```
{
  "Pyramid.Depth": 3,
  "Pyramid.NTilesX": 13,
  "Pyramid.NTilesY": 13,
  "Pyramid.PixelSize": 0.115,
  "Pyramid.PixelsX": 3328,
  "Pyramid.PixelsY": 3328,
  "Pyramid.TileSize": 256,
  "Pyramid.x0": 3333.4320185778092,
  "Pyramid.y0": 43115.88078877725
}
```

**opening spooled pcs in dh5view, adding supertile module to recipe with level set to 2**
```
recipes.activeRecipe.namespace['supertile'].mdh
<NestedClassMDHandler>:

Pyramid.TileSize: 256
Pyramid.x0: 20703312.7
Pyramid.y0: 103083012.8
Pyramid.PixelSize: 0.115
Pyramid.Depth: 4
Pyramid.NTilesX: 0
Pyramid.NTilesY: 0
Pyramid.PixelsX: 0
Pyramid.PixelsY: 0
voxelsize.x: 0.46
voxelsize.y: 0.46
```

**abridged tiled pcs metadata**
```
voxelsize.x: 0.115
voxelsize.y: 0.115
voxelsize.z: 0.2
voxelsize.units: um
Camera.ROI: [180, 896, 436, 1152]
Camera.ROIHeight: 256
Camera.ROIOriginX: 180
Camera.ROIOriginY: 896
Camera.ROIWidth: 256
CameraOrientation.FlipX: False
CameraOrientation.FlipY: False
CameraOrientation.Rotate: False
Multiview.ActiveViews: [0]
Multiview.ChannelColor: [0, 1, 1, 0]
Multiview.NumROIs: 4
Multiview.ROI0Origin: [180, 896]
Multiview.ROI1Origin: [744, 896]
Multiview.ROI2Origin: [1144, 896]
Multiview.ROI3Origin: [1684, 896]
Multiview.ROISize: [256, 256]
Positioning.x: 3500.0
Positioning.x_piezo: 3500.0
Positioning.x_target: 3500.0
Positioning.y: 43200.0
Positioning.y_piezo: -43200.0
Positioning.y_target: 43200.0
Positioning.z: 49.981500000000004
Positioning.z_piezo: 49.979800000000004
Positioning.z_target: 50.0
Protocol.DataStartsAt: 0
```

noting that the off parts are not due to the supertile source, but just tile_pyramid alone gets it
```
from PYME.Analysis import tile_pyramid
from tempfile import TemporaryDirectory
x, y = tile_pyramid.get_position_from_events(image.events, image.mdh)
p = tile_pyramid.tile_pyramid(TemporaryDirectory(), image.data, x, y, image.mdh, pyramid_tile_size=256)
p.mdh
<NestedClassMDHandler>:

Pyramid.TileSize: 256
Pyramid.x0: 20703312.7
Pyramid.y0: 103083012.8
Pyramid.PixelSize: 0.115
Pyramid.Depth: 4
Pyramid.NTilesX: 0
Pyramid.NTilesY: 0
Pyramid.PixelsX: 0
Pyramid.PixelsY: 0
```

**Comment by barentine** [comment]
<details> <summary> x, y are good </summary>
```
x(np.arange(image.data.shape[2]))
array([3500. , 3500. , 3500. , 3500. , 3520.8, 3511.2, 3491.3, 3479.4,
       3486.4, 3505.9, 3520. , 3541.6, 3536.5, 3522.5, 3502.9, 3482.7,
       3466.6, 3458.8, 3461. , 3472.8, 3491.2, 3511.8, 3529.5, 3540. ,
       3562.5, 3559. , 3549.1, 3533.7, 3514.7, 3494. , 3474. , 3456.9,
       3444.5, 3438.2, 3438.7, 3446. , 3459.2, 3476.9, 3497.1, 3517.7,
       3536.3, 3551. , 3560. , 3583.3, 3580.7, 3573.1, 3560.9, 3545. ,
       3526.3, 3505.9, 3485.2, 3465.3, 3447.7, 3433.3, 3423. , 3417.6,
       3417.2, 3422. , 3431.7, 3445.6, 3462.9, 3482.4, 3503.1, 3523.6,
       3542.6, 3559. , 3571.7, 3580. , 3583.2, 3604.1, 3602. , 3595.9,
       3585.9, 3572.5, 3556.2, 3537.7, 3517.7, 3497. , 3476.4, 3456.7,
       3438.7, 3423.2, 3410.8, 3401.9, 3397. , 3396.1, 3399.4, 3406.7,
       3417.7, 3432. , 3449. , 3468. , 3488.3, 3509.1, 3529.5, 3548.8,
       3566.1, 3580.7, 3592.2, 3599.9, 3603.7, 3624.9, 3623.2, 3618. ,
       3609.6, 3598.2, 3584. , 3567.5, 3549.1, 3529.4, 3508.8, 3488. ,
       3467.6, 3448. , 3429.9, 3413.7, 3399.9, 3388.9, 3381. , 3376.3,
       3375.1, 3377.4, 3383. , 3391.9, 3403.8, 3418.4, 3435.2, 3453.8,
       3473.7, 3494.3, 3515.1, 3535.4, 3554.8, 3572.7, 3588.5, 3601.9,
       3612.5, 3619.9, 3624.1, 3645.7, 3644.2, 3639.8, 3632.5, 3622.6,
       3610.1, 3595.4, 3578.7, 3560.5, 3541. , 3520.7, 3499.9, 3479.2,
       3458.8, 3439.4, 3421.1, 3404.5, 3389.8, 3377.3, 3367.4, 3360.1,
       3355.7, 3354.3, 3355.8, 3360.2, 3367.5, 3377.5, 3390. , 3404.8,
       3421.4, 3439.7, 3459.2, 3479.5, 3500.3, 3521. , 3541.3, 3560.8,
       3579. , 3595.7, 3610.3, 3622.8, 3632.7, 3639.9, 3644.3, 3666.5,
       3665.2, 3661.4, 3655. , 3646.2, 3635.1, 3621.9, 3606.8, 3590. ,
       3571.8, 3552.5, 3532.4, 3511.8, 3491. , 3470.3, 3450.1, 3430.7,
       3412.4, 3395.4, 3380. , 3366.6, 3355.2, 3346.1, 3339.3, 3335.1,
       3333.5, 3334.4, 3338. , 3344. , 3352.5, 3363.3, 3376.3, 3391.1,
       3407.7, 3425.7, 3444.9, 3464.9, 3485.5, 3506.3, 3527. , 3547.2,
       3566.8, 3585.3, 3602.5, 3618. , 3631.7, 3643.4, 3652.9, 3659.9,
       3664.5, 3666.4, 3687.4, 3686.2, 3682.7, 3677. , 3669.2, 3659.2,
       3647.2, 3633.5, 3618.1, 3601.2, 3583.1, 3564. , 3544.1, 3523.6,
       3502.9, 3482.1, 3461.5, 3441.4, 3422. , 3403.6, 3386.4, 3370.6,
       3356.4, 3343.9, 3333.4, 3324.9, 3318.6, 3314.5, 3312.7, 3313.3,
       3316.1, 3321.2, 3328.5, 3337.9, 3349.3, 3362.6, 3377.5, 3394. ,
       3411.8, 3430.7, 3450.4, 3470.7, 3491.4, 3512.2, 3532.9, 3553.1,
       3572.7, 3591.4, 3609. , 3625.2, 3639.9, 3652.9, 3663.9, 3673. ,
       3679.9, 3684.6, 3687. ])
y(np.arange(image.data.shape[2]))
array([43200. , 43200. , 43200. , 43200. , 43200. , 43217.5, 43218.9,
       43202.9, 43184.2, 43180. , 43194.2, 43200. , 43220. , 43235. ,
       43241.5, 43237.9, 43224.9, 43205.9, 43185.4, 43168.5, 43159.3,
       43160.1, 43170.6, 43188.4, 43200. , 43220.4, 43238.6, 43252.6,
       43260.7, 43262.2, 43256.8, 43245.2, 43228.6, 43208.8, 43188.1,
       43168.7, 43152.7, 43142. , 43137.6, 43140.1, 43149.2, 43163.9,
       43182.6, 43200. , 43220.6, 43239.9, 43256.8, 43270.1, 43279. ,
       43283.1, 43281.9, 43275.7, 43264.8, 43249.8, 43231.8, 43211.8,
       43191. , 43170.8, 43152.4, 43137. , 43125.5, 43118.6, 43116.8,
       43120.2, 43128.5, 43141.3, 43157.7, 43176.7, 43197.2, 43200. ,
       43220.7, 43240.5, 43258.8, 43274.7, 43287.6, 43297. , 43302.6,
       43304. , 43301.4, 43294.6, 43284.2, 43270.3, 43253.7, 43234.9,
       43214.7, 43193.9, 43173.4, 43153.9, 43136.3, 43121.2, 43109.3,
       43101. , 43096.6, 43096.3, 43100.2, 43108. , 43119.6, 43134.3,
       43151.6, 43170.9, 43191.4, 43200. , 43220.7, 43240.9, 43259.9,
       43277.2, 43292.4, 43305.1, 43314.8, 43321.4, 43324.6, 43324.3,
       43320.6, 43313.6, 43303.4, 43290.3, 43274.8, 43257.1, 43237.9,
       43217.6, 43196.9, 43176.2, 43156.2, 43137.4, 43120.3, 43105.5,
       43093.2, 43084. , 43077.9, 43075.2, 43076. , 43080.2, 43087.8,
       43098.4, 43111.9, 43127.8, 43145.7, 43165.1, 43185.5, 43200. ,
       43220.7, 43241.1, 43260.6, 43278.8, 43295.5, 43310.2, 43322.6,
       43332.6, 43339.8, 43344.3, 43345.7, 43344.2, 43339.8, 43332.5,
       43322.5, 43310. , 43295.3, 43278.7, 43260.4, 43240.9, 43220.6,
       43199.8, 43179.1, 43158.8, 43139.3, 43121. , 43104.4, 43089.7,
       43077.3, 43067.3, 43060.1, 43055.7, 43054.3, 43055.8, 43060.3,
       43067.6, 43077.6, 43090.1, 43104.8, 43121.5, 43139.8, 43159.3,
       43179.6, 43200. , 43220.8, 43241.2, 43261. , 43279.8, 43297.4,
       43313.5, 43327.8, 43340.1, 43350.3, 43358. , 43363.4, 43366.1,
       43366.3, 43363.9, 43358.9, 43351.4, 43341.6, 43329.6, 43315.5,
       43299.7, 43282.3, 43263.6, 43243.9, 43223.5, 43202.8, 43182. ,
       43161.5, 43141.6, 43122.6, 43104.8, 43088.5, 43074. , 43061.4,
       43050.9, 43042.9, 43037.2, 43034.1, 43033.6, 43035.7, 43040.3,
       43047.4, 43057. , 43068.7, 43082.5, 43098.1, 43115.4, 43133.9,
       43153.5, 43173.8, 43194.5, 43200. , 43220.8, 43241.3, 43261.3,
       43280.6, 43298.8, 43315.9, 43331.5, 43345.5, 43357.7, 43367.9,
       43376.1, 43382.1, 43385.9, 43387.3, 43386.5, 43383.4, 43378. ,
       43370.4, 43360.7, 43349. , 43335.5, 43320.3, 43303.6, 43285.7,
       43266.7, 43246.8, 43226.4, 43205.7, 43184.9, 43164.3, 43144.1,
       43124.6, 43106.1, 43088.7, 43072.7, 43058.2, 43045.5, 43034.7,
       43025.9, 43019.3, 43014.9, 43012.8, 43013. , 43015.6, 43020.3,
       43027.3, 43036.5, 43047.6, 43060.6, 43075.4, 43091.7, 43109.3,
       43128. , 43147.6, 43167.9, 43188.6])
```
</details>

issue is I botched the nm -> um in the offset
```
xps.min() + image.mdh.voxelsize_nm.x * 1e3 * x0_cam
20703312.7
xps.min() + image.mdh.voxelsize_nm.x / 1e3 * x0_cam
3333.3999999999996

```

**Comment by barentine** [comment]
OK, still something left to debug I assume
![image](https://user-images.githubusercontent.com/31105780/101408909-15b9e700-38ab-11eb-85b9-89c6b283f364.png)
some swap was being used. All of this is interesting since we make it in /tmp and the drive still has 75 G available, plus what's written looks rather small
![image](https://user-images.githubusercontent.com/31105780/101409448-ebb4f480-38ab-11eb-9d65-aa6119a8972c.png)

**Comment by barentine** [comment]
long and short of it is I'd thought that `x0` in tiler was being used to set `self._x0` used in `Tick` and hadn't caught that these were not the same.

The current `tiler` code adds tiles relative to the stage x0/y0 un-referenced to the chip. This works because we add the pyramid x0 and y0 onto tile_coords in SupertileDatasource.

At the moment in `tile_pyramid.tile_pyramid` I botched our effort to make things chip relative by trying to make the pyramid chip-relative - I missed that we add x0 and y0 on in the SupertileDatasource and leave the ImagePyramid referenced to its minimum x and minimum y positions. This makes sense, else we'd make ImagePyramids with more tiles than they really need, and as such `add_base_tile` is only functional for positive positions:
```
tile_xs = range(int(np.floor(x / self.tile_size)), int(np.floor((x + frameSizeX) / self.tile_size) + 1))
tile_ys = range(int(np.floor(y / self.tile_size)), int(np.floor((y + frameSizeY) / self.tile_size) + 1))
self.n_tiles_x = max(self.n_tiles_x, max(tile_xs))
self.n_tiles_y = max(self.n_tiles_y, max(tile_ys))
```

and my trying to make the ImagePyramid chip origin relative results in us always getting some negative positions, since I subtract off not just the min x and min y, but also the offset to the roi on the chip
```
xdp
array([1449, 1449, 1449, 1630, 1546, 1373, 1270, 1330, 1500, 1623, 1810,
       1766, 1644, 1474, 1298, 1158, 1090, 1110, 1212, 1372, 1551, 1705,
       1797, 1992, 1962, 1876, 1742, 1577, 1397, 1223, 1074,  966,  911,
        916,  979, 1094, 1248, 1423, 1603, 1764, 1892, 1970, 2173, 2150,
       2084, 1978, 1840, 1677, 1500, 1320, 1147,  994,  869,  779,  732,
        729,  770,  855,  976, 1126, 1296, 1476, 1654, 1819, 1962, 2072,
       2144, 2172, 2354, 2336, 2283, 2196, 2079, 1937, 1777, 1603, 1423,
       1243, 1072,  916,  781,  673,  596,  553,  545,  574,  637,  733,
        857, 1005, 1170, 1347, 1528, 1705, 1873, 2023, 2150, 2250, 2317,
       2350, 2535, 2520, 2475, 2402, 2303, 2179, 2036, 1876, 1704, 1525,
       1344, 1167,  997,  839,  698,  578,  483,  414,  373,  363,  383,
        431,  509,  612,  739,  885, 1047, 1220, 1399, 1580, 1757, 1925,
       2081, 2218, 2335, 2427, 2491, 2528, 2716, 2703, 2664, 2601, 2515,
       2406, 2278, 2133, 1975, 1805, 1629, 1448, 1268, 1090,  922,  763,
        618,  490,  382,  296,  232,  194,  182,  195,  233,  297,  383,
        492,  621,  765,  924, 1094, 1270, 1451, 1631, 1808, 1977, 2136,
       2281, 2408, 2517, 2603, 2665, 2703, 2897, 2885, 2852, 2797, 2720,
       2623, 2509, 2377, 2231, 2073, 1905, 1730, 1551, 1370, 1190, 1015,
        846,  687,  539,  405,  289,  190,  110,   51,   15,    1,    9,
         40,   92,  166,  260,  373,  502,  646,  803,  970, 1143, 1323,
       1503, 1683, 1859, 2030, 2190, 2340, 2475, 2594, 2696, 2778, 2839,
       2879, 2896, 3078, 3068, 3037, 2988, 2920, 2833, 2729, 2610, 2476,
       2329, 2171, 2005, 1832, 1654, 1474, 1293, 1114,  939,  770,  610,
        461,  323,  200,   91,    0,  -74, -129, -164, -180, -175, -150,
       -106,  -43,   39,  138,  254,  383,  527,  682,  846, 1017, 1194,
       1374, 1555, 1735, 1910, 2081, 2243, 2397, 2537, 2665, 2778, 2874,
       2953, 3013, 3054, 3075], dtype=int32)
```


**so**
I'd submit a new PR to fix it, but I think best is to edit #665 since I'll be editing/adding some doc and we can avoid a merge conflict that way
